### DISCUSSION TITLE
Should the alias parameter in the Path method be removed?

### DISCUSSION THREAD

**Discussion by bowenerchen** [fastapi/fastapi#12330] [Questions] [answered]
### Privileged issue

- [ ] I'm @tiangolo or he asked me directly to create an issue here.

### Issue Content

This is my MRE code:
```python
from __future__ import annotations

from typing import Annotated

import uvicorn
from fastapi import FastAPI, Path

app = FastAPI()


@app.get('/{item_id}')
async def async_root(item_id: Annotated[int, Path(title = 'Item ID', alias = 'xid')]):
    return {'item_id': item_id}


if __name__ == '__main__':
    uvicorn.run(app, host = '127.0.0.1', port = 18081)
```
and I got the swagger docs:
http://127.0.0.1:18081/docs#/default/async_root__item_id__get
![QQ_1727667100103](https://github.com/user-attachments/assets/fbc42966-eba6-4f96-aa4f-bb85243022f7)

Question:
1. Why is the content of the title field not displayed in the documentation?
2. 
![QQ_1727667237322](https://github.com/user-attachments/assets/22d006f8-7ddc-4175-b7d9-1da86b55628b)
The generated curl request is incorrect; the placeholder {item_id} has not been replaced with the value 123 for xid
3. When I try to manually send a curl request to access it, I still get an error:
```python
curl http://127.0.0.1:18081/123
{"detail":[{"type":"missing","loc":["path","xid"],"msg":"Field required","input":null}]}
```
4. When I remove the alias field from the Path method, it works normally：
```python
@app.get('/{item_id}')
async def async_root(item_id: Annotated[int, Path(title = 'Item ID')]):
    return {'item_id': item_id}
```
![QQ_1727667506471](https://github.com/user-attachments/assets/d64fa7d2-e704-4ccd-82cf-37505b37c043)
```python
 curl http://127.0.0.1:18081/123
{"item_id":123}
```
the `alias` field description:
![QQ_1727667820560](https://github.com/user-attachments/assets/1f8eb781-0092-4189-96a9-4e24397ec520)

My Environment:
Python 3.12.4
FastAPI 0.115.0
macOS M2 Max 14.6.1
PyCharm 2024.2.3 (Professional Edition)

**Comment by sinisaos** [comment, marked as answer]
@bowenerchen I think `alias` is useful for passing a Python reserved keyword or similar as a path parameter (as noted in the FastAPI docs). You must pass an `alias` to a path operation decorator like this
```python
@app.get("/{xid}") # <- here
async def async_root(item_id: Annotated[int, Path(alias="xid")]):
    return {"item_id": item_id}
```
After that everything works. 
```bash
> curl http://127.0.0.1:18018/123
{"item_id":123}
```
Sorry if I missed your point.

**Reply by bowenerchen** [reply]
It seems that I used it incorrectly
I misunderstood the correct usage of the alias parameter
Is there specific documentation that guides this usage? 
I may not have seen that part yet

Sorry so much!!!
